rust/src/clib/lib.rs

```diff
@@ -167,7 +170,7 @@ pub extern "C" fn nmstate_net_state_apply(
 
     net_state.set_timeout(rollback_timeout);
 
-    // TODO: save log to the output pointer
+    let _ = env_logger::try_init();
```

Comment:
Ahh that's the logger I was looking for, thanks! I will try to integrate it.

Reply:
@cathay4t Looks like the tokio-rs projects has a tracing library with non-global tracer https://github.com/tokio-rs/tracing, you can define the scope of a trace and run code there.

Maybe  nmstate rust can use that library don't know if we can make it use memory or retrieve what has being logged.

Reply:
The tokio tracing seems overkill for this project. Please try memory logger. But this can be done in another PR. For this PR, just remove the print in stdout.